libnavigation-core/src/test/java/com/mapbox/navigation/core/history/HistoryRecorderControllerTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.mapbox.navigation.core.history
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.navigation.base.options.HistoryRecorderOptions
+import com.mapbox.navigation.base.options.NavigationOptions
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import java.io.File
+import java.util.concurrent.CountDownLatch
+
+@RunWith(RobolectricTestRunner::class)
+class HistoryRecorderControllerTest {
+
+    private val context: Context = ApplicationProvider.getApplicationContext()
+    private val navigationOptionsBuilder = NavigationOptions.Builder(context)
+    private val logger: Logger = mockk(relaxUnitFun = true)
+
+    // How do you mockk native methods?
```

Comment:
You need to wrap the callable methods in an interface and provide the instance with a factory that's also mocked.

Reply:
Hmm dang, this kind of sucks for the `HistoryReader`. We'll need an instrumentation test for it then

Reply:
We can wrap native `HistoryReader` in a platform interface and provide it via factory to run unit tests against it.

Reply:
Something like this might get the job done:
```
internal class PlatformHistoryReader(private val historyReader: HistoryReader) {
    fun next() = historyReader.next()
}

internal object HistoryFactory {
    fun createPlatformHistoryReader(path: String) = PlatformHistoryReader(HistoryReader("path"))
}

@Test
fun test() {
    mockkObject(HistoryFactory)
    every { HistoryFactory.createPlatformHistoryReader() } returns mockk()
}
```

Reply:
agreeing here just getting around to it. think we should follow up with a ticket here

there are lots of places in the SDK that are not tested, calling native functions. I'm finding that we're inventing a new pattern on the fly and it would be best to think about this more

Reply:
Capturing here https://github.com/mapbox/mapbox-navigation-android/issues/4492